### PR TITLE
fix unreliable channel

### PR DESCRIPTION
**Pull request by Irev-Dev**
What you had seemed fine to me @paultag, I don't see why moving the callback up into the other class would be a problem so I've put @lf94 as a reviewer too.

Added a playw test for this too (tests hover which uses the unreliable channel).

Resolves https://github.com/KittyCAD/modeling-app/issues/2326

### CODE REVIEW

File: src/lang/std/engineConnection.ts [hunk 1]
```diff
@@ -1366,14 +1361,11 @@ export class EngineCommandManager {
...
-      otherEventCallbacks[localUnsubscribeId] = callback
-    } else {
-      this.subscriptions[event] = {
-        [localUnsubscribeId]: callback,
-      }
+    if (!this.subscriptions[event]) {
+      this.subscriptions[event] = {}
     }
+    this.subscriptions[event][localUnsubscribeId] = callback
+
```
Comment (Irev-Dev) [inline src/lang/std/engineConnection.ts:1368]: This and same with `subscribeToUnreliable` below is no change to functionality, but Paul had changed it and it is neater.

**Review by paultag — APPROVED** [review]
unsurprisingly no objections :)

**Comment by lf94** [comment]
FWIW this will need to be translated into my PR over here https://github.com/KittyCAD/modeling-app/pull/2105 (which I'm super sorry has been taking too long to fixup.) 

Otherwise everything looks good. Moving the method out of the class is binding an event sooner.

**Comment by Irev-Dev** [comment]
Oh okay, So do you mind @lf94 if I merge this then?

If you're okay with that, when you go to translate it into #2105 then it should be obvious if it's all working okay or not since I added an e2e test.

**Comment by lf94** [comment]
Yep yep merge :)

**Pull request merged**